pkg/operator/controllers/node/node_controller_test.go

```diff
@@ -54,6 +79,15 @@ func TestReconciler(t *testing.T) {
 			},
 			featureFlag: true,
 			wantErr:     `nodes "nonexistent-node" not found`,
+			wantConditions: []operatorv1.OperatorCondition{defaultAvailable, defaultProgressing,
+				{
+					Type: ControllerName + "Controller" + operatorv1.OperatorStatusTypeDegraded,
+					// Status:             operatorv1.ConditionTrue,
```

Comment:
should this be uncommented, along with all the `cnds.Degraded.Status = operatorv1.ConditionTrue` in the error flows in `node_controller.go`?

Reply:
Good catch. I think, as a part of this spike, we have determined that we should _not_ be propagating Degraded statuses to the ARO Cluster Operator, as that would impact cluster upgradeability. 

It is already the case that any controller Degraded=True statuses will _not_ propagate to the CO in this PR change (https://github.com/Azure/ARO-RP/blob/d5453e724abe3ddd8e5c3597fa8c2eb4ffd924dc/pkg/operator/controllers/clusteroperatoraro/clusteroperatoraro_controller.go#L89). 

As a result, we have two options:
- Do not set any controller Degraded=True statuses at all, and delete all of the error flows in `node_controller.go` that do so. 
- Do set these statuses (uncommenting the code), leaving those statuses and errors available to view on the ARO Cluster resource.

Reply:
ok, if our goal is to not prevent upgrades when our own operator is degraded, then I agree with removing the commented out code.

Reply:
Discussing this with @SudoBrendan, we decided that we will set Controller degraded statuses and ensure they do not propagate to the cluster operator. I've enabled the Degraded=false handling on errors in the Node controller now.